doc-changelog/action.yml

```diff
@@ -0,0 +1,138 @@
+name: >
+  Documentation Changelog
+
+description: |
+  Adds a newsfragment to the target repository using ``towncrier``
+  containing the title and number of the pull request.
+
+
+inputs:
+
+  # Required inputs
+
+  path-to-fragment:
+    description: >
+      Path to where the towncrier fragment is stored
+    required: true
+    type: string
+
+  pr-branch:
+    description: >
+      Name of the pull request
+    required: true
+    type: string
+
+  pr-title:
+    description: >
+      Title of the pull request
+    required: true
+    type: string
+
+  pr-number:
+    description: >
+      The pull request number
+    required: true
+    type: string
+
+  token:
+    description: >
+      GitHub token for doing a git commit & push
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.pr-branch }}
+
+    - name: "Set up Python ${{ inputs.python-version }}"
+      uses: ansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: ${{ inputs.use-python-cache }}
+
+    - name: "Install towncrier"
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip towncrier
+
+    - name: "Get pull request type"
+      shell: bash
+      run: |
+        # Check inputs.pr-title is surrounded by double quotes
+        title='${{ inputs.pr-title }}'
+        if [[ "$title" =~ ^\".*\"$ ]]; then
+          echo "PR title is wrapped in double quotes"
+        else
+          echo "PR title is not wrapped in double quotes"
+          title="${{ inputs.pr-title }}"
+        fi
+
+        # Save title into env variable
+        echo "PR-TITLE=$title" >> $GITHUB_ENV
+
+        # Check if pr-title string starts with Add, Remove, Change, Fix, or Bump
+        added=`if [[ $title =~ ^[a,A]dd ]]; then echo "True"; fi`
+        removed=`if [[ $title =~ ^[r,R]emove ]]; then echo "True"; fi`
+        changed=`if [[ $title =~ ^[c,C]hange ]]; then echo "True"; fi`
+        fixed=`if [[ $title =~ ^[f,F]ix ]]; then echo "True"; fi`
+        dependency=`if [[ $title =~ ^[b,B]ump ]]; then echo "True"; fi`
```

Comment:
Ok.... shall we really use the PR title?? the PR branch name?? or the PR labels?? Being aware the labels tends to be applied by the bot in many cases.

I'm in for the labels. It is easy to change... many times you start a branch with a purpose and ends for something else.

Reply:
I think it would be best to make sure the PR title follows certain naming conventions (like in this action https://github.com/ansys/actions/blob/release/4.1/commit-style/action.yml) because if someone names it incorrectly, they can just fix it and re-run the action